doc: update documentation with sync. printk limitations in Matter (#14171)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/known_issues.rst

.. rst-class:: v2-5-2 v2-5-1 v2-5-0 v2-4-3 v2-4-2 v2-4-1 v2-4-0 v2-3-0 v2-2-0 v2-1-4 v2-1-3 v2-1-2 v2-1-1

> KRKNWK-18316: When the :kconfig:option:`CONFIG_PRINTK_SYNC` option is enabled in Matter over Thread application, the IEEE 802.15.4 radio driver may calculate invalid IEEE 802.15.4 radio frame timestamps.

Reviewer: ```suggestion
KRKNWK-18316: When the :kconfig:option:`CONFIG_PRINTK_SYNC` Kconfig option is enabled in a Matter over Thread application, the IEEE 802.15.4 radio driver may calculate invalid IEEE 802.15.4 radio frame timestamps
```

>   This is caused because the implementation of synchronous ``printk`` in Zephyr uses ``spinlock`` synchronization primitive which can block Real Time Clock interrupts that are needed by radio driver to calculate precise timestamps.

Reviewer: ```suggestion
  This is caused by the implementation of synchronous ``printk`` in Zephyr using ``spinlock`` synchronization primitive, which can block Real Time Clock interrupts that are needed by the radio driver to calculate precise timestamps.
```

>   **Workaround:** If enabled, disable :kconfig:option:`CONFIG_PRINTK_SYNC` in your application.

Reviewer: ```suggestion
  **Workaround:** If it is enabled, disable the :kconfig:option:`CONFIG_PRINTK_SYNC` Kconfig option in your application.
```
